pkg/cpio/newc/newc.go

```diff
@@ -142,6 +152,9 @@ func (w *writer) WriteRecord(f cpio.Record) error {
 		return err
 	}
 
+	if hdr.FileSize == 0 {
```

Comment:
yeah. The problem was that when hdr.FileSize is zero, the io.Reader is too, and kaboom!